core/src/main/java/org/elasticsearch/cluster/routing/RoutingNodes.java

```diff
@@ -47,6 +40,12 @@
  */
 public class RoutingNodes implements Iterable<RoutingNode> {
 
+    private final static Random RANDOM;
+
+    static {
+        RANDOM = Randomness.create();
+    }
+
```

Comment:
if we do that, lets do the 'looking' and set a static final boolean so it does not impact real code.